### PR TITLE
Introduce netty-codec-protobuf, netty-codec-marshalling and netty-codec-compression maven modules

### LINKED ISSUE #14239: Increase modularity of netty-codec
Some packages of `netty-codec` are self contained in their package and are not used elsewhere in `the netty-codec` and fall in a situtation very much like `netty-xml-codec`.

These packages could be extracted to their own module with the only impact of users adding the right dependency to their dependency descriptors.

- `io.netty.handler.codec.protobuf`
- `io.netty.handler.codec.marshalling`

It seems like a good opportunity also to decouple `netty-codec` from these less core libraries and increase the modularity of the project.

The `io.netty.handler.codec.compression` seems also to be a good candidate, it is reused by HTTP codecs and is more core-ish to Netty. If that's the case it would be added as a dependency of the HTTP modules.

### PR DESCRIPTION
**Pull request by vietj**
Motivation:

The `netty-codec` maven module contains a three silos packages (`protobuf`/`marshalling`/`compression`) which pull non-Netty optional dependencies. These silos can be extracted without impacting the `netty-codec` maven module. Each package becomes then an independant maven module each having its dependencies. This also facilitates the introdution of java modules and increase the modularity of the Netty project.

Changes:

The following maven modules are added `netty-codec-protobuf`, `netty-codec-marshalling` and `netty-codec-compression`, corresponding code/resources/dependencies are moved from `netty-codec` to each target. Native image and revapi declarations are kept up to date as well. HTTP codecs now depends on `netty-codec-compression` in addition of `netty-codec`.

Result:

The `netty-codec` is decoupled from `protobuf`/`marshalling`/`compression`, project modularity has increased. The `protobuf` and `marshalling` project external dependencies are not anymore optional.

Fixes #14239

### CODE REVIEW

File: codec-compression/pom.xml [hunk 1]
```diff
@@ -0,0 +1,147 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2012 The Netty Project
```
Comment (chrisvest) [inline codec-compression/pom.xml:3]: This should be 2024 since this pom file is new.

File: codec-marshalling/pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2012 The Netty Project
```
Comment (chrisvest) [inline codec-marshalling/pom.xml:3]: 📌

File: codec-protobuf/pom.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2012 The Netty Project
```
Comment (chrisvest) [inline codec-protobuf/pom.xml:3]: 📌

**Review by chrisvest — APPROVED** [review]
Small license header comments, and the pom file has a merge conflict, but otherwise looks good.

**Comment by vietj** [comment]
@chrisvest will change that right now

File: testsuite-osgi/pom.xml [hunk 4]
```diff
@@ -240,6 +258,7 @@
             <configuration>
               <name>%bundle.symbolicName%.link</name>
               <value>%url%</value>
+              <separator>${line.separator}</separator>
```
Comment (normanmaurer) [inline testsuite-osgi/pom.xml:261]: why this change ?

Comment (vietj) [reply testsuite-osgi/pom.xml:261]: without this change, the plugin fails because it needs to aggregate several dependencies and fails because the line separarator is not configured

https://github.com/veithen/alta/blob/5e528a17fae44285d3e446b9a9a12e1dc69525b3/src/main/java/com/github/veithen/alta/AbstractGenerateMojo.java#L276
